src/main/java/org/dependencytrack/tasks/scanners/TrivyAnalysisTask.java

```diff
@@ -0,0 +1,486 @@
+
+         try {
+            HttpUriRequest request = buildRequest(requestUrl, scan);
+
+            try (final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request))) {
+
+
+                if (response.getStatusLine().getStatusCode() >= HttpStatus.SC_OK && response.getStatusLine().getStatusCode() < HttpStatus.SC_MULTIPLE_CHOICES) {
+                    String responseString = EntityUtils.toString(response.getEntity());
+                    Gson gson = new Gson();
+                    var trivyResponse = gson.fromJson(responseString,TrivyResponse.class);
+
+                    LOGGER.debug("Scan response: " + response.getStatusLine().getStatusCode());
+                    LOGGER.debug("Response from server: " + responseString);
+                    return trivyResponse;
+                } else {
+                    handleUnexpectedHttpResponse(LOGGER, request.getURI().toString(), response.getStatusLine().getStatusCode(), response.getStatusLine().getReasonPhrase());
+                }
+            }
+        } catch (Throwable  ex) {
+            handleRequestException(LOGGER, ex);
+        }
+        return null;
+    }
+
+    private void deleteBlob(PutRequest input) {
+        final String requestUrl = "%s/twirp/trivy.cache.v1.Cache/DeleteBlobs".formatted(apiBaseUrl);
+        DeleteRequest delete = new DeleteRequest();
+        delete.setBlobIDS(new String[] {input.getDiffID()});
+
+         try {
+            HttpUriRequest request = buildRequest(requestUrl, delete);
+            final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request));
+            LOGGER.debug("DeleteBlob response: " + response.getStatusLine().getStatusCode());
+        } catch (Throwable  ex) {
+            handleRequestException(LOGGER, ex);
+        }
+
+    }
+
+    private void handle(final Component component, final org.dependencytrack.parser.trivy.model.Vulnerability data) {
+        if (component == null) {
+            LOGGER.error("Unable to handle null component");
+            return;
+        } else if (data == null) {
+            addNoVulnerabilityToCache(component);
+            return;
+        }
+
+        try (QueryManager qm = new QueryManager()) {
+            final var trivyParser = new TrivyParser();
+
+            Vulnerability parsedVulnerability = trivyParser.parse(data, qm);
+            final Component componentPersisted = qm.getObjectByUuid(Component.class, component.getUuid());
+
+            if (componentPersisted != null && parsedVulnerability.getVulnId() != null) {
+                Vulnerability vulnerability = qm.getVulnerabilityByVulnId(parsedVulnerability.getSource(), parsedVulnerability.getVulnId());
+
+                if (vulnerability == null) {
+                    LOGGER.warn("Vulnerability not available:" + parsedVulnerability.getSource()  + " - " + parsedVulnerability.getVulnId());
```

Comment:
Can we reduce this to `debug` level? Perhaps also with a bit more info, e.g. `... will create it now` so users don't have to wonder whether *they* need to do something.

Reply:
done